### ISSUE TITLE
Successful logon from IP and failure from a different IP rule doesn't work with ipv6

### ISSUE THREAD

**Issue by OmarEfat** [Azure/Azure-Sentinel#8280]
**Describe the bug**
This rule produces false positive if the successful and failure logons from the same ipv6 address 
**To Reproduce**
1. Go to 'https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Azure%20Active%20Directory/Analytic%20Rules/SuccessThenFail_DiffIP_SameUserandApp.yaml'
2. Scroll down to line 31

**Expected behavior**
Users had a successful logons and failure logons do not show up as a result of the query.

**Possible solution**
This possibly can solve the issue since the ipv6 are encrypted and can't be checked if they are hosted in the same network so we will need to use the whole address?
**SuccessIPBlock = iff(IPAddress contains ":", IPAddress, strcat(split(IPAddress, ".")[0], ".", split(IPAddress, ".")[1]))**

Please let us know how to update this one.

**Comment by v-vdixit** [comment]
Hi @OmarEfat thanks for flagging this, we are working with the team on this will update you shortly.

**Comment by v-vdixit** [comment]
Hi @OmarEfat we are waiting for update from the concerned team, will update you by the end of this week, thanks!

**Comment by OmarEfat** [comment]
@v-vdixit Okay sounds good!

**Comment by v-vdixit** [comment]
Hi @OmarEfat we have received a suggestion that we should compare first two parts of IPv6 also and not the entire IP, can you please check and confirm if that works as a resolution for you, thanks!

**Comment by OmarEfat** [comment]
Hi @v-vdixit 
Can you confirm that this is what you mean
` SuccessIPBlock = iff(IPAddress contains ":", strcat(split(IPAddress, ":")[0], ":", split(IPAddress, ":")[1]), strcat(split(IPAddress, ".")[0], ".", split(IPAddress, ".")[1])),`

I guess so though. I don't see any identical Success and Failed IP addresses.

Thanks for your assistance.

**Comment by v-vdixit** [comment]
Yes @OmarEfat this is similar to what we are referring, thanks, please let us know if this resolves your issue, thanks!

**Comment by OmarEfat** [comment]
Looks good, thanks @v-vdixit

**Comment by jms1015hw** [comment]
Hi @OmarEfat, can I confirm this resolved your issue?

I am having a similar concern...

**Comment by v-vdixit** [comment]
Hi @OmarEfat please confirm if we can close on this issue, thanks!

**Comment by OmarEfat** [comment]
Hi @jms1015hw Yes, it solved the problem.

Hi @v-vdixit Okay I'm confirming that, thanks for your help.
